packages/next-swc/crates/core/src/relay.rs

```diff
@@ -73,120 +84,50 @@ impl Fold for Relay {
 }
 
 #[derive(Debug)]
-enum BuildRequirePathError<'a> {
+enum BuildRequirePathError {
     FileNameNotReal,
-    MultipleSourceSetsFound {
-        source_set_names: Vec<SourceSetName>,
-        path: &'a PathBuf,
-    },
-    ProjectNotFoundForSourceSet {
-        source_set_name: SourceSetName,
-    },
-    FileNotASourceFile,
-    CouldNotCategorize {
-        err: Cow<'static, str>,
-        path: String,
-    },
 }
 
-// This is copied from https://github.com/facebook/relay/blob/main/compiler/crates/relay-compiler/src/build_project/generate_artifacts.rs#L251
-// until the Relay team exposes it for external use.
 fn path_for_artifact(
     root_dir: &Path,
     source_path: &Path,
-    project_config: &ProjectConfig,
+    config: &Config,
     definition_name: &str,
 ) -> PathBuf {
-    let source_file_location_key = SourceLocationKey::Standalone {
-        path: source_path.to_str().unwrap().parse().unwrap(),
-    };
-    let filename = if let Some(filename_for_artifact) = &project_config.filename_for_artifact {
-        filename_for_artifact(source_file_location_key, definition_name.parse().unwrap())
-    } else {
-        match &project_config.typegen_config.language {
-            relay_config::TypegenLanguage::Flow => format!("{}.graphql.js", definition_name),
-            relay_config::TypegenLanguage::TypeScript => {
-                format!("{}.graphql.ts", definition_name)
-            }
+    let filename = match &config.language {
+        RelayLanguageConfig::Flow => format!("{}.graphql.js", definition_name),
+        RelayLanguageConfig::TypeScript => {
+            format!("{}.graphql.ts", definition_name)
         }
     };
 
-    let output_path = create_path_for_artifact(project_config, source_file_location_key, filename);
-
-    if project_config.output.is_some() {
-        let absolute_output_path = root_dir.join(&output_path);
-
-        let diffed_path =
-            diff_paths(&absolute_output_path, &source_path.parent().unwrap()).unwrap();
-
-        return diffed_path;
-    }
+    let output_path = if let Some(artifacts_directory) = &config.artifacts_directory {
+        root_dir.join(artifacts_directory).join(filename)
+    } else {
+        root_dir
+            .join(source_path)
+            .parent()
+            .unwrap()
+            .join("__generated__")
+            .join(filename)
+    };
 
```

Comment:
Btw, I think we can probably just error in this case, saying that Next.js will not support `relay` config without `artifactDirectory` outside of `pages`.

Reply:
Yeah that makes more sense to me.